packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/CustomVisConfigTabs.tsx

```diff
@@ -65,6 +157,257 @@ const loadMonaco = (monaco: Monaco, schemas: Schema[]) => {
+                    Loading a new template will overwrite your current chart
+                    configuration. Are you sure you want to continue?
+                </Text>
+                <Group position="right" mt="sm">
+                    <Button
+                        color="dark"
+                        variant="outline"
+                        onClick={() => {
+                            setSelectedTemplate(undefined);
+                        }}
+                    >
+                        Keep config
+                    </Button>
+                    <Button
+                        onClick={() => {
+                            if (selectedTemplate)
+                                loadTemplate(selectedTemplate);
+                            setSelectedTemplate(undefined);
+                        }}
+                    >
+                        Load template
+                    </Button>
+                </Group>
+            </Modal>*/}
+
+            <Select
+                label="Load vega lite template"
+                placeholder="Select template"
+                data={Object.values(TemplateType)}
+                onChange={(value) => {
+                    if (!value) return;
+                    // if (isEditorEmpty) {
+                    loadTemplate(value as TemplateType);
+                    // } else {
+                    //   setSelectedTemplate(value as TemplateType);
+                    //}
+                }}
+            />
+        </>
+    );
+};
+const GenerateVizWithAi = ({
+    itemsMap,
+    sampleResults,
+    editorConfig,
+    setEditorConfig,
+}: {
+    itemsMap: ItemsMap | undefined;
+    sampleResults: {
+        [k: string]: unknown;
+    }[];
+    editorConfig: string;
+    setEditorConfig: (config: string) => void;
+}) => {
+    const { projectUuid } = useParams<{
+        projectUuid: string;
+    }>();
+    const [prompt, setPrompt] = useState('');
+
+    const [isLoading, setIsLoading] = useState(false);
+
+    const { showToastError } = useToaster();
+
+    const handleSubmit = useCallback(() => {
+        if (isLoading) return;
+
+        setIsLoading(true);
+        if (prompt && projectUuid)
+            getCustomViz(
+                projectUuid,
+                prompt,
+                itemsMap,
+                sampleResults,
+                editorConfig,
+            )
+                .then((vizConfig) => {
+                    // Handle the visualization config
+                    setEditorConfig(vizConfig);
+                })
+                .catch((error) => {
+                    showToastError({
+                        title: 'Error generating custom viz with AI',
+                        subtitle: getErrorMessage(error),
+                    });
+                    console.error('Error generating custom viz:', error);
+                })
+                .finally(() => {
+                    setIsLoading(false);
+                });
+    }, [
```

Comment:
this should be a react query mutation instead

Reply:
updated